astropy/io/registry/base.py

```diff
@@ -187,10 +188,17 @@ def delay_doc_updates(self, cls):
         because the documentation of the corresponding ``read`` and ``write``
         methods are build every time.
 
+        This context manager has no effect when running Python in optimized mode
+        (-OO CLI flag).
+
         Examples
         --------
         see for example the source code of ``astropy.table.__init__``.
         """
+        if sys.flags.optimize >= 2:
```

Comment:
Do we not have a CI test for optimized python?

Reply:
Not yet: https://github.com/astropy/astropy/pull/17472